cmake/CUDA_utils.cmake

```diff
@@ -35,6 +35,26 @@ function(adjust_source_file_language_property SOURCES)
   endif()
 endfunction()
 
+# Generate list of xx-real for every specified supported architecture.
+# List should be sorted in increasing order.
+#
+# The last one will also be repeated as xx-virtual to ensure the generation of PTX for most recent
+# virtual architecture and maintain forward compatibility.
+function(CUDA_get_cmake_cuda_archs out_args_list arch_values)
```

Comment:
I don't like the fact, that we have mixed global level code and function declarations here - I see I did it also myself when introducing `adjust_source_file_language_property`, but this `CUDA_get_cmake_cuda_archs` is somewhat connected to the code below the listing and setting of default architectures and the rest of code is below - it made the diff a bit surprising to read.